feat(cli-mock): add section for searchable mocking (#4775)
aws-amplify/docs: src/pages/cli/usage/mock.mdx

- [@primaryKey, @index](/cli/graphql/data-modeling)
- [@hasOne, @hasMany, @belongsTo, @manyToMany](/cli/graphql/data-modeling)
- [@function](/cli/graphql/custom-business-logic#lambda-function-resolver)
- [@searchable](/cli/graphql/search-and-result-aggregations/)

> ### Mocking the lambda triggers on models

Reviewer: ```suggestion
### Mocking the Lambda triggers on @model types
```

In addition, you can use a `.env` file within the function directory (ie. `<project root>/amplify/backend/function/<function name>/.env`) to override any environment variables for local mocking.

> ### Mocking the searchable models

Reviewer: ```suggestion
### Mocking @model types with @searchable
```

> If you use `@searchable` directives on `@model` types in your GraphQL schema, then you can test the search GraphQL queries that are generated for you locally via `amplify mock` or `amplify mock api`.

Reviewer: ```suggestion
If you use the `@searchable` directive on `@model` types in your GraphQL schema, then you can test the search GraphQL queries that are generated for you locally via `amplify mock` or `amplify mock api`.
```

> To learn more about the search queries, refer [this guide](/cli/graphql/search-and-result-aggregations/).

Author: Do you think we need an example GraphQL search query here? Different types of search queries are present in the guide I linked here. Would that suffice?

> To learn more about the search queries, refer [this guide](/cli/graphql/search-and-result-aggregations/).

Reviewer: ```suggestion
To learn more about the search queries, refer to [this guide](/cli/graphql/search-and-result-aggregations/).
```

We create the following artifacts when you mock the API with searchable models:

> - [Opensearch](https://opensearch.org/) version `1.3.0` is downloaded.

Reviewer: ```suggestion
- [OpenSearch](https://opensearch.org/) version `1.3.0` is downloaded.
```

> - A python lambda trigger is stored locally at `mock-api-resources/searchable/searchable-lambda-trigger`. 

Reviewer: ```suggestion
- A python Lambda trigger is stored locally at `mock-api-resources/searchable/searchable-lambda-trigger`. 
```

> This lambda trigger is automatically invoked locally if you perform a CRUD operation on the searchable model types. 

Reviewer: ```suggestion
This Lambda trigger is automatically invoked locally if you perform a CRUD operation on the searchable model types. 
```

> It updates the data in the corresponding search index used by the locally running opensearch cluster.

Reviewer: ```suggestion
It updates the data in the corresponding search index used by the local OpenSearch cluster.
```

> - A data folder to store the opensearch indices is created at `mock-api-resources/searchable/searchable-data`.

Reviewer: ```suggestion
- A data folder to store the OpenSearch indices is created at `mock-api-resources/searchable/searchable-data`.
```
